### ISSUE TITLE
Feature Request: Use Chroma instead of Pinecone

### ISSUE THREAD

**Issue by bjorkmeister** [pashpashpash/vault-ai#6]
Rationale: I would like to set this up locally and Pinecone only has a paid subscription model, if you are not whitelisted through the waiting list.

I would like to set this up using a local running version of Chroma.

**Comment by Armadillidiid** [comment]
Weaviate would be nice too. It's free and open source

**Comment by bradfair** [comment]
+1 for abstraction at the storage layer; I'd like to use Milvus.

**Comment by Jasmin25** [comment]
> +1 Don't know which service I'd use but since pinecone is basically paid only for the time being, I'd love to use "anything else". Don't mind setting it up myself.

+1

**Comment by isConic** [comment]
Pinecone has a free-tier that works for Vault-ai. People can use it without paying.

**Comment by Bergdoktor** [comment]
Yeah, but there's a waiting list for the free tier. I actually just got in
yesterday after waiting for about 5 days.

isConic ***@***.***> schrieb am Fr., 21. Apr. 2023, 03:25:

**Comment by botheredbybees** [comment]
I'd like to run this against our technical and scientific documentation on a blue water research vessel with limited internet bandwidth, so the 'anything else' option would good to see.

**Comment by barretto94** [comment]
> Pinecone has a free-tier that works for Vault-ai. People can use it without paying.

I'm not too keen on the user agreement where you:
_Grant Pinecone and its suppliers a non-exclusive, royalty-free license to access, use, reproduce, modify and display the Customer Data for the purposes of (i) providing the Hosted Services, (ii) exercising its rights and obligations under this Agreement; (iii) generating Aggregated and Anonymous Data (as defined below).... Customer further agrees that Pinecone shall own such Aggregated and Anonymous Data and may retain, use and disclose such data for any lawful business purpose, including to improve its products and services._

For proof of concept that's okay, but with sensitive data, I rather host that in-house, a secured data center, or cloud host.

**Comment by zakariaelh** [comment]
same here, I don't see a reason to host in pinecone when running it locally. Also Pinecone charges by time of use (no by how many times you access the data) and it's very expensive.
